index.js

```diff
@@ -1,6 +1,57 @@
 /* eslint-env node */
 'use strict';
 
+const path = require('path');
+const VersionChecker = require('ember-cli-version-checker');
+
+function requireTransform(transformName) {
+  let plugin = require(transformName);
+
+  plugin = plugin.__esModule ? plugin.default : plugin;
+
+  // adding `baseDir` ensures that broccoli-babel-transpiler does not
+  // issue a warning and opt out of caching
+  let pluginPath = require.resolve(`${transformName}/package`);
+  let pluginBaseDir = path.dirname(pluginPath);
+  plugin.baseDir = () => pluginBaseDir;
+
+  return plugin;
+}
+
 module.exports = {
-  name: 'ember-decorators'
+  name: 'ember-decorators',
+
+  included(app) {
+    this._super.included.apply(this, arguments);
+
+    const disableTransforms = app.options.emberDecorators && app.options.emberDecorators.disableTransforms;
```

Comment:
Should use something like https://github.com/babel/ember-cli-babel/blob/master/index.js#L137 to get the options. 

As it is this will not work for nested addons wanting to use decorators.